tests/jobs/auto_correlate_observations/test_integration.py

```diff
@@ -0,0 +1,159 @@
+    )
+
+    test_data_dir = os.path.join(TEST_DATA_DIR, "local", "snake_oil")
+
+    shutil.copytree(test_data_dir, "test_data")
+    os.chdir(os.path.join("test_data"))
+
+    res_config = ResConfig("snake_oil.ert")
+    ert = EnKFMain(res_config)
+
+    config = {"clustering": {"spearman_correlation": {"fcluster": {"t": 1.0}}}}
+    config_file = "my_config_file.yaml"
+    with open(config_file, "w") as f:
+        yaml.dump(config, f)
+
+    auto_correlate_observations.AutoCorrelateObservationsJob(ert).run(config_file)
+
+    # call_args represents the clusters, we expect the snake_oil
+    # observations to generate this amount of them
+    # call_args is a call object, which itself is a tuple of args and kwargs.
+    # In this case, we want args, and the first element of the arguments, which
+    # again is a tuple containing the configuration which is a list of configs.
+    assert len(run_mock.call_args[0][0]) == 47, "wrong number of clusters"
+
+
+@pytest.mark.skipif(TEST_DATA_DIR is None, reason="no libres test-data")
+@pytest.mark.usefixtures("setup_tmpdir")
+def test_auto_correlate_main_entry_point_no_config(monkeypatch):
+    run_mock = Mock()
+    scal_job = Mock(return_value=Mock(run=run_mock))
+    monkeypatch.setattr(
+        auto_correlate_observations, "CorrelatedObservationsScalingJob", scal_job,
+    )
+
+    test_data_dir = os.path.join(TEST_DATA_DIR, "local", "snake_oil")
+
+    shutil.copytree(test_data_dir, "test_data")
+    os.chdir(os.path.join("test_data"))
+
+    res_config = ResConfig("snake_oil.ert")
+    ert = EnKFMain(res_config)
+
+    auto_correlate_observations.AutoCorrelateObservationsJob(ert).run()
+
+    assert len(run_mock.call_args[0][0]) > 1
+
+
+@pytest.mark.skipif(TEST_DATA_DIR is None, reason="no libres test-data")
+@pytest.mark.usefixtures("setup_tmpdir")
+def test_auto_correlate_with_scaling():
+    test_data_dir = os.path.join(TEST_DATA_DIR, "local", "snake_oil")
+
+    shutil.copytree(test_data_dir, "test_data")
+    os.chdir(os.path.join("test_data"))
+
+    res_config = ResConfig("snake_oil.ert")
+    ert = EnKFMain(res_config)
+
+    config = {"clustering": {"spearman_correlation": {"fcluster": {"t": 1.0}}}}
+    config_file = "my_config_file.yaml"
+    with open(config_file, "w") as f:
+        yaml.dump(config, f)
+
+    auto_correlate_observations.AutoCorrelateObservationsJob(ert).run(config_file)
+
+    # assert that this arbitrarily chosen cluster gets scaled as expected
+    obs = ert.getObservations()["FOPR"]
+    for index in [13, 14, 15, 16, 17, 18, 19, 20]:
+        assert obs.getNode(index).getStdScaling() == 2.8284271247461903
+
+
+@pytest.mark.skipif(TEST_DATA_DIR is None, reason="no libres test-data")
+@pytest.mark.usefixtures("setup_tmpdir")
+def test_auto_correlate_skip_clusters_yielding_empty_data_matrixes(monkeypatch):
+    def raising_scaling_job(data):
+        if data == {"CALCULATE_KEYS": {"keys": [{"index": [88, 89], "key": "FOPR"}]}}:
```

Comment:
This one would benefit from a comment or two, since it's hard to see what's happening.